ui/components/app/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js

```diff
@@ -1,19 +1,48 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { getTxData, getUnapprovedTransactions } from '../../../../selectors';
+import { transactionMatchesNetwork } from '../../../../../shared/modules/transaction.utils';
+import { I18nContext } from '../../../../contexts/i18n';
+import { CONFIRM_TRANSACTION_ROUTE } from '../../../../helpers/constants/routes';
+import { clearConfirmTransaction } from '../../../../ducks/confirm-transaction/confirm-transaction.duck';
+import { hexToDecimal } from '../../../../../shared/lib/metamask-controller-utils';
 
-const ConfirmPageContainerNavigation = (props) => {
-  const {
-    onNextTx,
-    totalTx,
-    positionOfCurrentTx,
-    nextTxId,
-    prevTxId,
-    showNavigation,
-    firstTx,
-    lastTx,
-    ofText,
-    requestsWaitingText,
-  } = props;
+const ConfirmPageContainerNavigation = () => {
+  const t = useContext(I18nContext);
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const unapprovedTxs = useSelector(getUnapprovedTransactions);
+  const txData = useSelector(getTxData);
+
+  const network = hexToDecimal(txData.chainId);
+
```

Comment:
The "Navigate transactions - should confirm and remove an unapproved transaction" e2e test is failing because: `TimeoutError: Waiting for element to be located By(xpath, .//*[contains(concat(' ', normalize-space(./@class), ' '), ' confirm-page-container-navigation ')][(contains(string(.), '1 of 3') or contains(string(.), '1of3'))])`

This probably means that during the e2e test run `showNavigation` is false, which probably means that `enumUnapprovedTxs` is an empty array. Looking at this new code for getting  `currentNetworkUnapprovedTxs` compared to the old code in `confirm-transaction-base.container.js`, I notice that the `chainId` and `network` are here derived from the transaction but in `confirm-transaction-base.container.js` they are taken directly from `state.metamask`.

If this feature is working when running it manually in the browser, for both custom approvals and erc-20 token transfers, then the problem is likely in e2e test fixtures (unless there is something wrong with getting the chainId and network from the transaction?... I don't think so, but worth a double check)

Reply:
You are right @danjm. Good catch! 
I fixed that and also made a small change that solved the problem, so now `txData` is being passed from a parent component.
You can check it here: 1d005ae893d9a9da7350b0d14c5f649ac9905858

Reply:
why did you have to change it so that `txData` is being passed in, instead of being selected? what problem did that solve?

Reply:
When a user tries to confirm one of the transactions, it now shows the navigation, but doesn't get updated as it should - `txData` object being selected becomes empty, next `tx` cannot be seen on the next page and `currentPosition` is `-1` as the `currentTx` cannot be found from `enumUnapprovedTxs`, because there is no `txId` from `txData` object which is empty at that moment.
Screencast:

https://user-images.githubusercontent.com/92310504/208698663-50a774d2-29f0-4d4c-b161-7732f0e26efa.mov

Reply:
txData is derived from state in confirm-transaction-base.container.js which means we can get it here without drilling the prop down the line

Reply:
I get it, but the problem with it is that somehow the `txData` gets empty when one of the transactions is confirmed.
You can see it in the above attached video where `txDataFromSelector` is actually the transaction data derived from state. I `console.log`-ged it in order to show this problem...